packages/runtime/src/widgets/Form/Radio.tsx

```diff
@@ -38,19 +98,55 @@ export const RadioWidget: React.FC<RadioWidgetProps> = (props) => {
     [action],
   );
 
-  const formInstance = Form.useFormInstance();
+  // extract template data
+  const { namedData } = useTemplateData({
+    data: itemTemplate?.data,
+    name: itemTemplate?.name,
+  });
 
-  useEffect(() => {
-    setValue(values?.initialValue?.toString());
-  }, [values?.initialValue]);
+  const RadioComponent = useMemo(() => withTemplate(Radio), []);
 
-  useEffect(() => {
-    if (formInstance) {
-      formInstance.setFieldsValue({
-        [values?.id ?? values?.label ?? ""]: value,
+  // handle radio items
+  const radioItems = useMemo(() => {
+    const radioOptions: RadioOptionsProps[] = [];
+
+    const getStyle = (styles: EnsembleWidgetStyles): object => ({
+      ...styles,
+      ...(styles?.visible === false ? { display: "none" } : undefined),
+    });
+
+    if (values?.items) {
+      values.items.forEach((item) => {
+        radioOptions.push({
+          item,
+          children: item.label,
+          disabled: values.enabled === false || item.enabled === false,
+          value: String(item.value),
+          style: getStyle(item.styles as object),
+        });
       });
     }
-  }, [value, formInstance]);
+
+    if (isObject(itemTemplate) && !isEmpty(namedData)) {
+      map(namedData, (item: { [key: string]: unknown }) => {
+        const typedItem = get(item, itemTemplate.name) as CustomScope;
+        radioOptions.push({
+          item,
+          disabled: values?.enabled === false || typedItem.enabled === false,
```

Comment:
Don't check `typedItem` for enabled - it's data it can't be guaranteed to have enabled prop.